lib/src/date-input/Calendar.tsx

```diff
@@ -0,0 +1,488 @@
+            </PickerHeaderButton>
+            <HeaderMonthYear aria-live="polite">{innerDate.format("MMMM YYYY")}</HeaderMonthYear>
+            <PickerHeaderButton
+              aria-label="next month"
+              onClick={() => setInnerDate(innerDate.set("month", innerDate.get("month") + 1))}
+            >
+              <svg fill="currentColor" focusable="false" viewBox="0 0 24 24" aria-hidden="true">
+                <path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"></path>
+                <path fill="none" d="M0 0h24v24H0V0z"></path>
+              </svg>
+            </PickerHeaderButton>
+          </PickerHeader>
+          <WeekHeader>
+            {weekDays.map((weekDay) => (
+              <DayHeaderCell key={weekDay}>{weekDay}</DayHeaderCell>
+            ))}
+          </WeekHeader>
+
+          <DayCellsContainer>
+            {dayCells.map((day) =>
+              day !== 0 ? (
+                <DayCell
+                  onKeyDown={(event) => handleDayKeyboardEvent(event, day)}
+                  id={"day_" + day}
+                  aria-label={day}
+                  onClick={() => handleDateSelect(day, "date")}
+                  selected={day === innerDate.get("date") && selectedDate.isSame(innerDate, "day")}
+                  autoFocus={day === innerDate.get("date") && selectedDate.isSame(innerDate, "day")}
+                  aria-selected={day === innerDate.get("date") && selectedDate.isSame(innerDate, "day")}
+                >
+                  {day}
+                </DayCell>
+              ) : (
+                <EmptyDayCell />
+              )
+            )}
+          </DayCellsContainer>
+        </Calendar>
+      )}
+      {content === "yearPicker" && (
+        <YearPicker>
+          {getYearsArray().map((year) => (
+            <YearPickerButton
+              aria-label={year}
+              selected={selectedDate.get("year") === year}
+              autoFocus={selectedDate.get("year") === year}
+              onClick={() => {
+                handleDateSelect(year, "year");
+                setContent("calendar");
+              }}
+            >
+              <span>{year}</span>
+            </YearPickerButton>
+          ))}
+        </YearPicker>
+      )}
+      {content === "monthPicker" && (
+        <MonthPicker>
+          {months.map((month, index) => (
+            <MonthPickerButton
+              aria-label={month}
+              selected={selectedDate.get("month") === index}
+              autoFocus={selectedDate.get("month") === index}
+              onClick={() => {
+                handleDateSelect(index, "month");
+                setContent("calendar");
+              }}
+            >
+              {month}
+            </MonthPickerButton>
+          ))}
+        </MonthPicker>
+      )}
+    </DatePicker>
+  );
+};
+
+function getYearsArray() {
+  const yearList = [];
+  for (let i = firstYear; i <= lastYear; i++) {
+    yearList.push(i);
+  }
+  return yearList;
```

Comment:
Are you creating an array of years **each time** you open the year picker? I think this is not the best approach. 

Maybe you can create a stateful variable that depends on `date` prop and create the array from that value. Also, I think we can remove `firstYear` and `lastYear` variables, and calculate the years by a maximum and a minimum number of years to show.

Reply:
I agree that calculating the years by a maximum and a minimum number of years to show is a better approach, but we would need to discuss more this behaviour, right now the year selector works the same way as the one before.

In regards of creating the array each time I agree with you I will rethink this to avoid it.

Reply:
I think what Jiale did here is right. Since the function of `getYearsArray` is defined and executed outside of the main component, it is only calculated once, which is the correct approach!